src/ray/pubsub/subscriber.h

```diff
@@ -394,6 +405,9 @@ class Subscriber : public SubscriberInterface {
   std::shared_ptr<WaitForRefRemovedChannel> wait_for_ref_removed_channel_
       GUARDED_BY(mutex_);
 
+  /// ObjectLocationsChannel channel.
```

Comment:
I'm sorry, I think I probably missed asking it in the previous PR. But could you add comments here about why it needs to be a shared pointer?

Reply:
Hmm this sounds like too low level comments (makes no sense to justify all low level decisions in the comment). For the context, it is because it is shared by channel_

Reply:
Hmm, I don't think it's sharing semantics right? 
Btw, I'm asking this is because I'd want to make sure, we only use shared_ptr when it's actually sharing, and for sharing semantics, we'd like to specify it explicitly so that we can know better about the life cycle of this object, for example, when will this object be released.